File: DotNetMessaging/backend/DotNetMessaging.API/DTOs/MessageDTOs.cs
namespace DotNetMessaging.API.DTOs;

public class MessageDto
{
    public string Id { get; set; } = string.Empty;
    public string? ChatId { get; set; }
    public string? GroupId { get; set; }
    public string SenderId { get; set; } = string.Empty;
    public string SenderName { get; set; } = string.Empty;
    public string? SenderProfilePicture { get; set; }
    public string Content { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string? MediaUrl { get; set; }
    public string? MediaType { get; set; }
    public string? MediaFileName { get; set; }
    public long? MediaSize { get; set; }
    public string? ReplyToMessageId { get; set; }
    public MessageDto? ReplyToMessage { get; set; }
    public List<ReactionDto> Reactions { get; set; } = new();
    public DateTime CreatedAt { get; set; }
}

public class CreateMessageRequest
{
    public string? ChatId { get; set; }
    public string? GroupId { get; set; }
    public string Content { get; set; } = string.Empty;
    public string Type { get; set; } = "Text";
    public string? ReplyToMessageId { get; set; }
    public string? MediaUrl { get; set; }
    public string? MediaType { get; set; }
    public string? MediaFileName { get; set; }
    public long? MediaSize { get; set; }
}

public class ReactionDto
{
    public string Id { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public string UserName { get; set; } = string.Empty;
    public string Emoji { get; set; } = string.Empty;
}

public class AddReactionRequest
{
    public string MessageId { get; set; } = string.Empty;
    public string Emoji { get; set; } = string.Empty;
}

File: DotNetMessaging/SpringBootMessaging/src/main/java/com/messaging/api/dtos/MessageDTOs.java
package com.messaging.api.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

public class MessageDTOs {
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class MessageDto {
        private String id;
        private String chatId;
        private String groupId;
        private String senderId;
        private String senderName;
        private String senderProfilePicture;
        private String content;
        private String type;
        private String mediaUrl;
        private String mediaType;
        private String mediaFileName;
        private Long mediaSize;
        private String replyToMessageId;
        private MessageDto replyToMessage;
        private List<ReactionDto> reactions;
        private LocalDateTime createdAt;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class CreateMessageRequest {
        private String chatId;
        private String groupId;
        private String content;
        private String type = "Text";
        private String replyToMessageId;
        private String mediaUrl;
        private String mediaType;
        private String mediaFileName;
        private Long mediaSize;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ReactionDto {
        private String id;
        private String userId;
        private String userName;
        private String emoji;
    }
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class AddReactionRequest {
        private String messageId;
        private String emoji;
    }
}

